src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy

```diff
@@ -0,0 +1,175 @@
+        assert response.seatbid?.first()?.bid?.collect { it.price } == [getRoundedFloorValue(convertedMinFloorValue)]
+        assert response.cur == bidRequest.cur[0]
+    }
+
+    def "PBS should update bidFloor, bidFloorCur for signalling when floorMinCur is defined in request"() {
+        given: "BidRequest with floorMinCur"
+        def floorMin = randomFloorValue
+        def bidRequest = bidRequestWithFloors.tap {
+            cur = [EUR]
+            imp[0].bidFloor = floorMin
+            imp[0].bidFloorCur = EUR
+            ext.prebid.floors.floorMin = floorMin
+            ext.prebid.floors.floorMinCur = USD
+        }
+
+        and: "Account with enabled fetch, fetch.url in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.site.publisher.id)
+        accountDao.save(account)
+
+        and: "Set Floors Provider response with a currency different from the floorMinCur, floorValur lower then floorMin"
+        def floorProviderCur = EUR
+        def convertedMinFloorValue = floorMin *
+                getCurrencyRate(bidRequest.ext.prebid.floors.floorMinCur, floorProviderCur)
+
+        def floorsResponse = PriceFloorRules.priceFloorRules.tap {
+            floorMin = randomFloorValue
+            data.modelGroups[0].values = [(rule): convertedMinFloorValue - 0.1]
+            data.modelGroups[0].currency = floorProviderCur
+        }
+        floorsProvider.setResponse(bidRequest.site.publisher.id, floorsResponse)
+
+        when: "PBS processes auction request"
+        floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request bidFloor should correspond floorMin"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        verifyAll(bidderRequest) {
+            imp[0]?.bidFloor == getRoundedFloorValue(convertedMinFloorValue)
+            imp[0]?.bidFloorCur == floorProviderCur
+        }
+    }
+
+    def "PBS should not update bidFloor, bidFloorCur for signalling when currency conversion is not available"() {
+        given: "Pbs config with disabled conversion"
+        def pbsService = pbsServiceFactory.getService(floorsConfig +
+                ["currency-converter.external-rates.enabled": "false"])
+
+        and: "BidRequest with floorMinCur"
+        def floorMin = randomFloorValue
+        def bidRequest = bidRequestWithFloors.tap {
+            cur = [EUR]
+            imp[0].bidFloor = floorMin
+            imp[0].bidFloorCur = USD
+            ext.prebid.floors.floorMin = floorMin
+            ext.prebid.floors.floorMinCur = USD
+        }
+
+        and: "Account with enabled fetch, fetch.url in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.site.publisher.id)
+        accountDao.save(account)
+
+        and: "Set Floors Provider response with a currency different from the floorMinCur"
+        def floorsResponse = PriceFloorRules.priceFloorRules.tap {
+            data.modelGroups[0].values = [(rule): randomFloorValue]
```

Comment:
Is this really a random function? Seems like this ought to be based on the floorMin above?